ompi/mca/coll/xhc/README.md

```diff
@@ -70,213 +85,287 @@ hierarchy and its configuration.
-`1K`).
+	- Example: `--mca coll_xhc_chunk_size 16K`
 
-    - Example: `--mca coll_xhc_cico_max 1K`
+* Optional: If you wish to focus on latencies of small/medium size messages,
+you can try altering the cico-to-zcopy switchover point (MCA
+`coll_xhc_cico_max`, default `1K`).
+	
+	- Example: `--mca coll_xhc_cico_max 1K`
 
 * Optional: If your application is heavy in Broadcast calls and you suspect
 that specific ranks might be joining the collective with delay and causing
-others to stall waiting for them, you could try enabling dynamic leadership
-(`coll_xhc_dynamic_leader`), and seeing if it marks an improvement.
-
-    - Example: `--mca coll_xhc_dynamic_leader 1`
+others to stall waiting for them, try enabling dynamic leadership (MCA
+`coll_xhc_dynamic_leader`), and seeing if it makes an improvement. Please let
+us know if it does :-).
+	
+	- Example: `--mca coll_xhc_dynamic_leader 1`
 
 ### Example command lines
 
 *Assuming `PATH` and `LD_LIBRARY_PATH` have been set appropriately.*
 
 Default XHC configuration:  
-`$ mpirun --mca coll libnbc,basic,xhc --mca coll_xhc_priority 100 --bind-to core <application>`
+`$ mpirun --mca coll_xhc_priority 100 --bind-to core <application>`
 
 XHC w/ numa-sensitive hierarchy, chunk size @ 16K:  
-`$ mpirun --mca coll libnbc,basic,xhc --mca coll_xhc_priority 100 --mca coll_xhc_hierarchy numa --mca coll_xhc_chunk_size 16K --bind-to core <application>`
+`$ mpirun --mca coll_xhc_priority 100 --mca coll_xhc_hierarchy numa --mca coll_xhc_chunk_size 16K --bind-to core <application>`
 
 XHC with flat hierarchy (ie. none at all):  
-`$ mpirun --mca coll libnbc,basic,xhc --mca coll_xhc_priority 100 --mca coll_xhc_hierarchy node [--bind-to core] <application>`
+`$ mpirun --mca coll_xhc_priority 100 --mca coll_xhc_hierarchy node [--bind-to core] <application>`
 
-## Publications
+## Benchmarking
 
-1. **A framework for hierarchical single-copy MPI collectives on multicore nodes**,  
-*George Katevenis, Manolis Ploumidis, Manolis Marazakis*,  
-IEEE Cluster 2022, Heidelberg, Germany.  
-https://ieeexplore.ieee.org/document/9912729
+This section outlines some tips for benchmarking XHC and intra-node MPI
+collectives in general.
 
-## Contact
+### Micro-Benchmarks
 
-- George Katevenis (gkatev@ics.forth.gr)
-- Manolis Ploumidis (ploumid@ics.forth.gr)
+For our micro-benchmarking purposes, we have been using [OSU's microbenchmark
+suite](https://mvapich.cse.ohio-state.edu/benchmarks/). However, when
+micro-benchmarking intra-node collectives, there are some important details
+that one needs to look out for.
 
-Computer Architecture and VLSI Systems (CARV) Laboratory, ICS Forth
+**CPU Cache** An issue with the OSU micro-benchmarks is that they use the same
+buffer for each iteration without altering it. Since modern processors
+implicitly cache data, this can lead to false/unrealistic/unrepresentative
+results, given that actual real-world applications do not (usually/optimally!)
+perform duplicate operations.
```

Comment:
I've also observed that using empty buffers doesn't provide accurate latency measurements while benchmarking nt-buffer-transfer (Ref: https://github.com/openucx/ucx/pull/9408) in intra-node use cases.

I appreciate you sharing the modified version of OSU-benchmarks, I will try this out.

I noticed another aspect about OSU-benchmarks, which always use page-aligned buffers for latency measurement. This may not accurately represent real-world MPI applications that might use unaligned transfers, potentially suffering from different penalties and even substantially decreased performance on some AMD CPUs (https://lunnova.dev/articles/ryzen-slow-short-rep-mov/). Have you encountered similar findings in your experience?

Reply:
@gkatev Do you have a table with the performance improvement numbers for any HPC apps (or any new AI apps) with xHC for intra-node cases with details like ranksXthreads, mapping, binding, machine CPU details, etc ?

Reply:
Have you had a look in our paper "A framework for hierarchical single-copy MPI collectives on multicore nodes" @ Cluster 2022? It includes a few HPC app benchmarks with XHC. But there have been many improvements to XHC between the version in that paper and the one in this PR.

Reply:
I went through the paper and I see you are getting good improvements with PiSvM, miniAMR and CNTK. Have you had a chance to test the performance on WRF, NWCHEM, or OpenFOAM?

Reply:
Of those I've only tried OpenFOAM at some point. If I recall correctly, it didn't make all that vast use of collectives like Allreduce, Reduce, Bcast, Barrier. Do you know how heavy use of collectives WRF and NWCHEM do?

Reply:
The collective percentage should depend on the type of input used right? I don't have much idea about a particular input and the MPI percentage/individual collective percentage, an application expert might help I guess :)